### ISSUE TITLE
Fails to parse Constructor extended attribute with argument having default value

### ISSUE THREAD

**Issue by dontcallmedom** [w3c/webidl2.js#190]
The following IDL fragment is valid (AFAICT) according to the WebIDL spec:
```
[Constructor(boolean truth = false)] interface Foo {};
```
but webidl2.js throws an error:
`Got an error before parsing any named definition: Unexpected token in extended attribute argument list, line 1 (tokens: " = false)] interface")`

(found this while parsing the WebIDL from [Web Assembly JS API](https://webassembly.github.io/spec/js-api/)

**Comment by dontcallmedom** [comment]
(for sake of clarity, that error goes away when removing the default value piece (i.e. ` = false`)

**Comment by saschanaz** [comment]
Non-optional boolean argument cannot have a default value, so it should be:

```webidl
[Constructor(optional boolean truth = false)] interface Foo {};
```

The error message is confusing, though.

**Comment by dontcallmedom** [comment]
sorry, my example was bad in that way, but the bug exists also for non-boolean default value, e.g. `[Constructor(double v = 0)] interface Foo {};`

**Comment by dontcallmedom** [comment]
oh, but I guess what you're saying doesn't just apply to boolean arguments - only optional arguments can have default values. That makes sense.

**Comment by saschanaz** [comment]
>I guess what you're saying doesn't just apply to boolean arguments

You're right. I thought I wrote *non-optional argument*...

**Comment by dontcallmedom** [comment]
FWIW, I'm fine with having this is issue closed, unless you want to keep it open for the error message (in which case it should probably be retitled).
